scripts/scil_compute_local_tracking.py

```diff
@@ -232,59 +339,67 @@ def main():
-
-    max_steps = int(args.max_length / args.step_size) + 1
-    streamlines_generator = LocalTracking(
-        _get_direction_getter(args),
-        BinaryStoppingCriterion(mask_data),
-        seeds, np.eye(4),
-        step_size=vox_step_size, max_cross=1,
-        maxlen=max_steps,
-        fixedstep=True, return_all=True,
-        random_seed=args.seed,
-        save_seeds=args.save_seeds)
-
-    scaled_min_length = args.min_length / voxel_size
-    scaled_max_length = args.max_length / voxel_size
-
-    if args.save_seeds:
-        filtered_streamlines, seeds = \
-            zip(*((s, p) for s, p in tqdm_if_verbose(
-                streamlines_generator, verbose=args.verbose,
-                total=total_nb_seeds, miniters=int(total_nb_seeds / 100))
-                  if scaled_min_length <= length(s) <= scaled_max_length))
-        data_per_streamlines = {'seeds': lambda: seeds}
-    else:
-        filtered_streamlines = \
-            (s for s in tqdm_if_verbose(
-                streamlines_generator, verbose=args.verbose,
-                total=total_nb_seeds, miniters=int(total_nb_seeds / 100))
-             if scaled_min_length <= length(s) <= scaled_max_length)
-        data_per_streamlines = {}
-
-    if args.compress:
-        # Compressing. Threshold is in mm, but we are working in voxel space.
-        # Equivalent of sft.to_voxmm:  streamline *= voxres
-        # Equivalent of sft.to_vox: streamline /= voxres
-        voxres = np.asarray(odf_sh_img.header.get_zooms()[0:3])
-        filtered_streamlines = (
-            compress_streamlines(s * voxres,
-                                 args.compress) / voxres
-            for s in filtered_streamlines)
-
-    tractogram = LazyTractogram(lambda: filtered_streamlines,
-                                data_per_streamlines,
-                                affine_to_rasmm=seed_img.affine)
-
-    filetype = nib.streamlines.detect_format(args.out_tractogram)
-    reference = get_reference_info(seed_img)
-    header = create_tractogram_header(filetype, *reference)
-
-    # Use generator to save the streamlines on-the-fly
-    nib.streamlines.save(tractogram, args.out_tractogram, header=header)
+    if not args.use_gpu:
+        # LocalTracking.maxlen is actually the maximum length
+        # per direction, so we need to divide by 2.0 to limit
+        # the streamline to the right length
+        max_steps_per_direction = int(args.max_length / 2.0 / args.step_size)
+        # make streamline slightly longer to filter out streamlines
+        # exceeding max_length
+        max_steps_per_direction += 1
+
+        streamlines_generator = LocalTracking(
+            _get_direction_getter(args),
+            BinaryStoppingCriterion(mask_data),
+            seeds, np.eye(4),
+            step_size=vox_step_size, max_cross=1,
+            maxlen=max_steps_per_direction,
+            fixedstep=True, return_all=True,
+            random_seed=args.seed,
+            save_seeds=True)
```

Comment:
Does this make the tracking longer when user does not need the seeds?

Reply:
I wouldn't think so, it just returns it instead of droping it. No additionnal operations are required.